crates/uv-build/src/lib.rs

```diff
@@ -849,20 +867,48 @@ async fn run_python_script(
     venv: &Virtualenv,
     script: &str,
     source_tree: &Path,
+    mut environment_variables: FxHashMap<String, String>,
 ) -> Result<Output, Error> {
+    // First check user supplied environment variables
+    let path = environment_variables.remove("PATH");
+    let os_path = env::var_os("PATH");
+    let modified_path = if let Some(user_path) = path {
+        match os_path {
+            // Prepend the user supplied PATH to the existing PATH
+            Some(env_path) => {
+                let user_path = PathBuf::from(user_path);
+                let new_path = env::split_paths(&user_path).chain(env::split_paths(&env_path));
+                env::join_paths(new_path).map_err(Error::BuildScriptPath)?
+            }
+            // Use the user supplied PATH
+            None => OsString::from(user_path),
+        }
+        // No user supplied PATH was given
+    } else {
+        // Check if we need to use the environment's PATH
+        match os_path {
+            Some(env_path) => env_path,
+            None => OsString::new(),
+        }
+    };
+
     // Prepend the venv bin dir to PATH
-    let new_path = if let Some(old_path) = env::var_os("PATH") {
-        let new_path = iter::once(venv.scripts().to_path_buf()).chain(env::split_paths(&old_path));
-        env::join_paths(new_path).map_err(Error::BuildScriptPath)?
+    let modified_path = if modified_path.len() > 0 {
+        let venv_path =
+            iter::once(venv.scripts().to_path_buf()).chain(env::split_paths(&modified_path));
+        env::join_paths(venv_path).map_err(Error::BuildScriptPath)?
     } else {
-        OsString::from("")
+        modified_path
     };
+
     Command::new(venv.python_executable())
         .args(["-c", script])
         .current_dir(source_tree.normalized())
         // Activate the venv
         .env("VIRTUAL_ENV", venv.root())
-        .env("PATH", new_path)
+        .env("PATH", modified_path)
+        // Pass in remaining environment variables
+        .envs(environment_variables)
```

Comment:
Could we put the `.envs(environment_variables)` before the `.env("VIRTUAL_ENV", venv.root())` to ensure `VIRTUAL_ENV` never gets overridden?